src/language_features/hover.rs

```diff
@@ -105,43 +116,112 @@ pub fn editor_hover(
                 MarkupKind::PlainText => contents.value,
             },
         },
     };
 
-    if contents.is_empty() && diagnostics.is_empty() && maybe_hover_modal.is_none() {
-        return;
-    }
+    match hover_type {
+        HoverType::Normal => {
+            if contents.is_empty() && diagnostics.is_empty() {
+                return;
+            }
 
-    let anchor_or_style = if let Some(hover_modal) = maybe_hover_modal.as_ref() {
-        contents = format!("{}\n---\n{}", hover_modal.context, contents);
-        "modal".to_string()
-    } else {
-        format!("{}", params.position)
+            let command = format!(
+                "lsp-show-hover {} %§{}§ %§{}§",
+                params.position,
+                contents.replace("§", "§§"),
+                diagnostics.replace("§", "§§"),
+            );
+            ctx.exec(meta, command);
+        }
+        HoverType::Modal {
+            modal_heading,
+            do_after,
+        } => {
+            show_hover_modal(meta, ctx, modal_heading, do_after, contents, diagnostics);
+        }
+        HoverType::InfoInHoverClient { fifo } => {
+            show_hover_in_hover_client(meta, ctx, fifo, contents);
+        }
     };
+}
 
-    let mut command = format!(
-        "lsp-show-hover {} %§{}§ %§{}§",
-        anchor_or_style,
+fn show_hover_modal(
+    meta: EditorMeta,
+    ctx: &Context,
+    modal_heading: String,
+    do_after: String,
+    contents: String,
+    diagnostics: String,
+) {
+    let contents = format!("{}\n---\n{}", modal_heading, contents);
+    let command = format!(
+        "lsp-show-hover modal %§{}§ %§{}§",
         contents.replace("§", "§§"),
-        diagnostics.replace("§", "§§")
+        diagnostics.replace("§", "§§"),
     );
-
-    // Wrap if we're using a HoverModal
-    if let Some(hover_modal) = maybe_hover_modal {
-        command = formatdoc!(
-            "eval %§
+    let wrapped_command = formatdoc!(
+        "eval %§
                  {}
                  {}
              §",
-            command.replace("§", "§§"),
-            hover_modal.do_after.replace("§", "§§")
-        );
+        command.replace("§", "§§"),
+        do_after.replace("§", "§§")
+    );
+    ctx.exec(meta, wrapped_command);
+}
+
+fn show_hover_in_hover_client(
+    meta: EditorMeta,
+    ctx: &Context,
+    hover_fifo: String,
+    contents: String,
+) {
+    if contents.is_empty() {
+        return;
     }
 
+    let handle = std::thread::spawn(move || {
+        // Note that we don't show diagnostics in the hover client
+        fs::write(hover_fifo, contents.as_bytes()).unwrap();
+    });
+
+    let command = formatdoc!(
+        "
+        %[
+            edit! -existing -readonly -fifo %opt[lsp_hover_fifo] *hover*
+            set buffer=*hover* filetype markdown
+        ]"
+    );
+
+    let client = meta.client.clone().unwrap_or_default();
+    let command = formatdoc!(
+        "
+        try %[
+            eval -client hoverclient {}
```

Comment:
This is a great improvement, which solves lots of long-standing issues.
A dedicated buffer is much more powerful than the info box.

In my quick planning over at https://github.com/kak-lsp/kak-lsp/issues/514 I suggested to use `%opt{docsclient}`, which is already used by `:doc`, `:man` and `:git`. However, since we want to (also) create this new client, your solution seems better.

I personally don't use `%opt{docsclient}` but I can imagine that someone would want this to always spawn in their docsclient.
Maybe it's worth adding a second attempt here: 

```kak
        try %[
            eval -client hoverclient {}
	] catch %[
            eval -client "%opt{docsclient}" {}
	] catch %[
	    new ...
	]
```

Not completely sure yet. We can also decide later.

One alternative is to have `lsp-hover` accept an optional client name as argument.
The only disadvantage I see is that it's less discoverable (you'd have to type `:lsp-hover` and read the docstring in the info box).

Reply:
I'm starting to think that the default should be to display the `*hover*` buffer in the current client like `:doc`, `:grep` and friends do (and if docsclient is set, use that instead). The current client is the lowest common denominator, it will work alright everywhere, also when tmux is not available (even with tmux, it depends on the window layout whether the new split fits in well -- it could be too small).

On top of that, probably keep the current behavior if a client name is given as argument,